### PR TITLE
 Add feature for .aggregate() and .sum()

### PR DESCRIPTION
**Pull request by Banandana**
Adds the changes required to implement .aggregate and .sum features to mquery

### CODE REVIEW

File: lib/mquery.js [hunk 1]
```diff
@@ -1731,6 +1801,76 @@ Query.prototype.find = function (criteria, callback) {
...
+  callback = this._wrapCallback('aggregate', callback, {
+      pipeline: pipe
+      , options: options
+  });
+
+  this._collection.aggregate(pipe, options, utils.tick(callback));
+  return this;
+}
+
+Query.prototype.sum = function (property, callback) {
```
Comment (vkarpov15) [inline lib/mquery.js]: Why does `sum()` actually execute the aggregation pipeline?

File: lib/mquery.js [hunk 2]
```diff
@@ -1731,6 +1801,76 @@ Query.prototype.find = function (criteria, callback) {
...
+Query.prototype.aggregate = function (pipeline, callback) {
+  this.op = 'aggregate';
+
+    if ('function' === typeof pipeline) {
+        callback = pipeline;
+        pipeline = undefined;
+    }
+    else if (Query.canMerge(pipeline))
+    {
+      this.merge(pipeline);
```
Comment (vkarpov15) [inline lib/mquery.js]: Won't this write to `_conditions`, not `_pipeline`?

**Comment by hexafluoride** [comment]
@vkarpov15 I don't blame you, this is supposed to be a private patch for a freelancing project, never designed to see the light of day. It doesn't implement every query condition, since there was a specific request and a list of conditions that needed to be implemented. And yes, the code quality is a mess. However, I hope this at least shows that implementing aggregate support and a translation layer from query conditions to an aggregation pipeline isn't as hard as it seems.

**Comment by kingkarki** [comment]
Any documentation on sum?

**Pull request closed without merging**